internal/app/networkd/pkg/networkd/netconf.go

```diff
@@ -45,3 +46,22 @@ func (n *NetConf) BuildOptions(config runtime.Configurator) error {
 
 	return nil
 }
+
+// BuildOptionsFromKernel translates the supplied config to functional options.
+func (n *NetConf) BuildOptionsFromKernel() error {
+	// Check to see if a kernel supplied configuration option was specified
+	kern := &address.Kernel{}
+	if err := kern.Discover(context.Background()); err != nil {
+		return nil
```

Comment:
While I agree that failing to build options from the kernel should not constitute an error, I think the error determination should be processed inside the discovery, since we should have situations where we genuinely encounter an error which should be handled (or at least logged), such as for a parse failure of the `ip=` clause, and those situations which should not constitute an error, such as the non-existence of an `ip=` clause, or for where `ip=dhcp`, perhaps.

Thus, it would seem to me a better path would be to use a sentinel error (e.g. `ErrNoop`) or return the parsed object or even a boolean flag to indicate whether to continue building the options.